examples/ticks_and_spines/date_concise_formatter.py

```diff
@@ -0,0 +1,182 @@
+"""
+================================================
+Formatting date ticks using ConciseDateFormatter
+================================================
+
+Finding good tick values and formatting the ticks for an axis that
+has date data is often a challenge.  `~.dates.ConciseDateFormatter` is
+meant to improve the strings chosen for the ticklabels, and to minimize
+the strings used in those tick labels as much as possible.
+
+.. note::
+
+    This formatter is a candidate to become the default date tick formatter
+    in future versions of Matplotlib.  Please report any issues or
+    suggestions for improvement to the github repository or mailing list.
+
+First, this is the default formatter.
+"""
+import datetime
+import matplotlib.pyplot as plt
+import matplotlib.dates as mdates
+import numpy as np
+
+# make a timeseries...
+base = datetime.datetime(2005, 2, 1)
+dates = np.array([base + datetime.timedelta(hours=(2 * i))
+                  for i in range(732)])
+N = len(dates)
+np.random.seed(19680801)
+y = np.cumsum(np.random.randn(N))
+
+fig, axs = plt.subplots(3, 1, constrained_layout=True, figsize=(6, 6))
+lims = [(np.datetime64('2005-02'), np.datetime64('2005-04')),
+        (np.datetime64('2005-02-03'), np.datetime64('2005-02-15')),
+        (np.datetime64('2005-02-03 11:00'), np.datetime64('2005-02-04 13:20'))]
+for nn, ax in enumerate(axs):
+    ax.plot(dates, y)
+    ax.set_xlim(lims[nn])
+    # rotate_labels...
+    for label in ax.get_xticklabels():
+        label.set_rotation(40)
+        label.set_horizontalalignment('right')
+axs[0].set_title('Default Date Formatter')
+plt.show()
+
+#############################################################################
+# The default date formater is quite verbose, so Matplotlib has an
+# alternative that can be chosen with a few extra lines of code.  Note,
```

Comment:
You should mention what the alternative is here; it may be in the code, but that doesn't mean it's obvious what's important.